mso/resource_mso_schema.go

```diff
@@ -50,18 +76,46 @@ func resourceMSOSchemaCreate(d *schema.ResourceData, m interface{}) error {
 	log.Printf("[DEBUG] Schema: Beginning Creation")
 	msoClient := m.(*client.Client)
 	name := d.Get("name").(string)
-	templateName := d.Get("template_name").(string)
-	tenandId := d.Get("tenant_id").(string)
+	tempVarTemplateName, ok_template_name := d.GetOk("template_name")
+	tempVarTemplates, ok_templates := d.GetOk("template")
 
-	schemaApp := models.NewSchema("", name, templateName, tenandId)
+	if !ok_template_name && !ok_templates {
+		return fmt.Errorf("template_name or a template block with its name, tenant_id and display_name are required.")
+	}
+
+	var schemaApp *models.Schema
+	if ok_template_name {
+		tempVarTenantId, ok := d.GetOk("tenant_id")
+		if !ok {
+			return fmt.Errorf("tenant_id is required when using template_name.")
+		}
+		templateName := tempVarTemplateName.(string)
+		tenantId := tempVarTenantId.(string)
+		schemaApp = models.NewSchema("", name, templateName, tenantId, make([]interface{}, 0, 1))
+
+	} else {
+		templates := make([]interface{}, 0, 1)
+		if ok_templates {
+			template_list := tempVarTemplates.(*schema.Set).List()
+			for _, val := range template_list {
+				map_templates := make(map[string]interface{})
+				inner_templates := val.(map[string]interface{})
+				map_templates["name"] = inner_templates["name"]
+				map_templates["displayName"] = inner_templates["display_name"]
+				map_templates["tenantId"] = inner_templates["tenant_id"]
+				templates = append(templates, map_templates)
+			}
+		}
+		schemaApp = models.NewSchema("", name, "", "", templates)
```

Comment:
Hi, @shrsr NewSchema function should expect 5 arguments and should return 2 values. But here in this case, we are only catching one of the return values.

Reply:
Same as above.